tools/tasks.py

```diff
@@ -460,11 +462,15 @@ def python_add_file_task():
             contents = infile.read()
```

Comment:
Would be worth spending a bit of time investigating to see if we can stream the file contents line-by-line, rather than calling `infile.read()`, which would load the whole log file into memory during redaction.
Given we can do that, we could also re-use the existing `redact_file()` function.

Reply:
Looking into it now. In terms of using the existing `redact_file()` function it does write out a file which in this case we don't need. We need to pass back gzip bytes?

Reply:
I could edit the `redact_string()` I made and take file as input and string as output but in the way its made I think you still need the entire gzip string to send back.

Reply:
Yeah, redact_file also couldn't really open it due to the fact its a gz...

Reply:
I agree with leaving this potential enhancement out of this PR to better isolate this fix.   @bbrks If you think this should be revisited, go ahead and file a followup ticket to track.